components/proxy/src/test/java/com/hotels/styx/proxy/StyxProxyTest.java

```diff
@@ -77,15 +76,14 @@ public void startsAndStopsAServer() {
     public void startsServerWithHttpConnector() {
         HttpInterceptor echoInterceptor = (request, chain) -> textResponse("Response from http connector");
 
-        int port = freePort();
         HttpServer server = NettyServerBuilder.newBuilder()
-                .setHttpConnector(connector(port))
+                .setHttpConnector(connector(0))
```

Comment:
I would say the code is a bit easier to write if we use freeport for the cases with  '0' and WireMockServer.dynamic*Port()  instead of having to find what's the correct way to get a random port in each case.  That doesn't mean we have to keep the 'port' variable used later...

Reply:
Hi @dvlato. `freePort` is problematic because it can easily cause port conflicts in parallel environments. Because it is susceptible for race conditions it should be avoided.